### ISSUE TITLE
[bug] Attribute error when accessing formatted value in layered config 

### ISSUE THREAD

**Issue by billcrook** [rochacbruno/dynaconf#482]
**Describe the bug**
Attribute error when accessing settings value in layered configuration with multiple settings files.

<details>
<summary>Exception</summary>

```
Traceback (most recent call last):
  File "app.py", line 11, in <module>
    assert settings['s3_url'] == expected_value # fails
  File "/home/user/dev/venv/lib/python3.7/site-packages/dynaconf/utils/functional.py", line 17, in inner
    return func(self._wrapped, *args)
  File "/home/user/dev/venv/lib/python3.7/site-packages/dynaconf/base.py", line 285, in __getitem__
    value = self.get(item, default=empty)
  File "/home/user/dev/venv/lib/python3.7/site-packages/dynaconf/base.py", line 419, in get
    data = (parent or self.store).get(key, default)
  File "/home/user/dev/venv/lib/python3.7/site-packages/dynaconf/utils/boxing.py", line 15, in evaluate
    value = f(dynabox, item, *args, **kwargs)
  File "/home/user/dev/venv/lib/python3.7/site-packages/dynaconf/utils/boxing.py", line 64, in get
    return super(DynaBox, self).get(item, default, *args, **kwargs)
  File "/home/user/dev/venv/lib/python3.7/site-packages/dynaconf/vendor/box/box.py", line 109, in get
    return B[C]
  File "/home/user/dev/venv/lib/python3.7/site-packages/dynaconf/utils/boxing.py", line 23, in evaluate
    return recursively_evaluate_lazy_format(value, settings)
  File "/home/user/dev/venv/lib/python3.7/site-packages/dynaconf/utils/__init__.py", line 355, in recursively_evaluate_lazy_format
    value = value(settings)
  File "/home/user/dev/venv/lib/python3.7/site-packages/dynaconf/utils/parse_conf.py", line 176, in __call__
    return self.formatter(self.value, **self.context)
  File "/home/user/dev/venv/lib/python3.7/site-packages/dynaconf/utils/parse_conf.py", line 138, in __call__
    return self.function(value, **context)
AttributeError: 'Settings' object has no attribute 's3_protocol'
```

</details>

**To Reproduce**

1. Having the following folder structure

<details>
<summary> Project structure </summary>

```bash
$ ls
.  ..  app.py  conf-test-default.toml  conf-test-layer.toml
```
</details>

2. Having the following config files:

<details>
<summary> Config files </summary>

**conf-test-default.toml**
```toml
s3_protocol = 's3a'
s3_url = '@format {this.s3_protocol}://{this.s3_bucket}'
```
and

**conf-test-layer.toml**
```toml
s3_bucket = 'kewl_bucket'
```

</details>

3. Having the following app code:

<details>
<summary> Code </summary>

**app.py**
```python
from dynaconf import Dynaconf

settings_files = ['conf-test-default.toml', 'conf-test-layer.toml']

settings = Dynaconf(settings_files=settings_files)

expected_value = 's3a://kewl_bucket'

assert settings.s3_url == expected_value # succeeds

assert settings['s3_url'] == expected_value # fails

assert settings.get('s3_url', default='s3://default') == expected_values # fails

assert settings('s3_url', cast=str) == expected_value # fails
```

</details>

4. Executing under the following environment

<details>
<summary> Execution </summary>

```bash
$ python app.py
```

</details>

**Expected behavior**
I would expect all documented settings access methods to function properly.

**Environment (please complete the following information):**
 - OS: Linux 5.9.10-arch1-1 #1 SMP PREEMPT Sun, 22 Nov 2020 14:16:59 +0000 x86_64 GNU/Linux
 - Dynaconf Version 3.1.2

**Comment by billcrook** [comment]
It seems that accessing via [__getitem__](https://github.com/rochacbruno/dynaconf/blob/327f5619a42a90fd12a180db2d4fa7811623b22e/dynaconf/base.py#L283) differs from via [__getattr__](https://github.com/rochacbruno/dynaconf/blob/327f5619a42a90fd12a180db2d4fa7811623b22e/dynaconf/base.py#L138). Specifically I see that  `__getattr__` calls `recursively_evaluate_lazy_format` whereas `__getitem__` does not.

**Comment by rochacbruno** [comment]
I think this issue was solved in #405 I will double check.

**Comment by billcrook** [comment]
:+1:
